File: src/api/library.js
import { get, post } from './client.js'

// GET /api/v1/library → {installed[], licenses[], subscriptions[]}
export const getLibrary = () => get('/library')

// POST /api/v1/library/install {appId} → LibraryEntryRead
export const install = (appId) => post('/library/install', { appId })

// POST /api/v1/library/uninstall {appId} → 200
export const uninstall = (appId) => post('/library/uninstall', { appId })

// POST /api/v1/library/subscribe {appId,planId} → LibraryEntryRead(kind=subscription)
export const subscribe = (appId, planId) => post('/library/subscribe', { appId, planId })

// POST /api/v1/library/buy-license {appId,planId} → LibraryEntryRead(kind=license)
export const buyLicense = (appId, planId) => post('/library/buy-license', { appId, planId })

File: backend/app/routers/library.py
"""资料库路由（单用户演示）。"""

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from ..crud.library import buy_license, get_library, install, subscribe, uninstall
from ..database import get_db
from ..schemas import LibraryActionIn, LibraryEntryRead, LibraryOut

router = APIRouter(prefix="/api/v1", tags=["library"])


@router.get("/library", response_model=LibraryOut)
def get_library_route(db: Session = Depends(get_db)):
    return get_library(db)


@router.post("/library/install", response_model=LibraryEntryRead)
def install_app(payload: LibraryActionIn, db: Session = Depends(get_db)):
    return install(db, payload.appId)


@router.post("/library/uninstall")
def uninstall_app(payload: LibraryActionIn, db: Session = Depends(get_db)):
    return uninstall(db, payload.appId)


@router.post("/library/subscribe", response_model=LibraryEntryRead)
def subscribe_app(payload: LibraryActionIn, db: Session = Depends(get_db)):
    return subscribe(db, payload.appId, payload.planId)


@router.post("/library/buy-license", response_model=LibraryEntryRead)
def buy_license_app(payload: LibraryActionIn, db: Session = Depends(get_db)):
    return buy_license(db, payload.appId, payload.planId)
